client/windows/CCastleInterface.cpp

```diff
@@ -1906,12 +1906,10 @@ const CBuilding * CFortScreen::RecruitArea::getMyBuilding()
 		return nullptr;
 
 	const CBuilding * build = town->town->buildings.at(myID);
-	while (town->town->buildings.count(myID))
-	{
-		if (town->hasBuilt(myID))
-			build = town->town->buildings.at(myID);
-		myID.advance(town->town->creatures.size());
-	}
+	myID = BuildingID::getDwellingFromLevel(level, 1);
+	if (town->hasBuilt(myID))
+		build = town->town->buildings.at(myID);
+
```

Comment:
I think it would show incorrect dwellings for pirates (3rd upgrade) in cove and for alternative creatures from tow

Reply:
Thanks. Will recheck later.